src/Lykke.AzureStorage/Blob/Decorators/RetryOnFailureAzureBlobDecorator.cs

```diff
@@ -108,5 +109,8 @@ public string GetBlobUrl(string container, string key)
 
         public async Task<IDictionary<string, string>> GetMetadataAsync(string container, string key) 
             => await _retryService.RetryAsync(async () => await _impl.GetMetadataAsync(container, key), _onGettingRetryCount);
+
+        public async Task SetPublicAccessPermissionsAsync(string container, BlobContainerPublicAccessType accessType)
+            => await _retryService.RetryAsync(async () => await _impl.SetPublicAccessPermissionsAsync(container, accessType), _onModificationsRetryCount);
```

Comment:
Can we omit async await for the lambda here?

Reply:
Yes, we can and not only in the lambda, but in the containing methods too. Fixed